src/DynamoPackages/PackageLoader.cs

```diff
@@ -386,16 +389,28 @@ public void LoadPackages(IEnumerable<Package> packages)
         }
 
         /// <summary>
-        /// Scans and compiles any new packages contained in new package paths that are added.
-        /// Note: This method will still reload ALL packages in the VM even those that are not new.
+        /// This method is called when custom nodes and packages need to be reloaded if there are new package paths.
         /// </summary>
         /// <param name="loadPackageParams"></param>
-        private void LoadNewPackages(LoadPackageParams loadPackageParams)
+        /// <param name="customNodeManager"></param>
+        public void LoadCustomNodesAndPackages(LoadPackageParams loadPackageParams, CustomNodeManager customNodeManager)
         {
+            foreach (var path in loadPackageParams.Preferences.CustomPackageFolders)
+            {
+                // Append the definitions subdirectory for custom nodes.
+                var dir = path == DynamoModel.BuiltInPackagesToken ? PathManager.BuiltinPackagesDirectory : path;
+                dir = TransformPath(dir, PathManager.DefinitionsDirectoryName);
+
+                customNodeManager.AddUninitializedCustomNodesInPath(dir, false, false);
```

Comment:
did we actually ever discuss loading the /dyf folder from the Built-In Packages folder? what do we need this for?

Reply:
I think that logic came out of this commit https://github.com/DynamoDS/Dynamo/commit/a9b67336d76d9a94e7736305572a7d2620fd3f52
Not sure about that @mjkkirschner 
@aparajit-pratap any ideas ?

Reply:
I think this is the discussion we had related to this: https://autodesk.slack.com/archives/C01C8GGT9EY/p1625175309318800